com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/RaytracingIntersection.hlsl

```diff
@@ -104,6 +104,11 @@ void FetchIntersectionVertex(uint vertexIndex, out IntersectionVertex outVertex)
 
     #ifdef ATTRIBUTES_NEED_COLOR
     outVertex.color      = UnityRayTracingFetchVertexAttribute4(vertexIndex, kVertexAttributeColor);
+
+    // We want to default to white in case there is no specified color, to match the raster behaviour
+    if (!any(outVertex.color))
```

Comment:
hmm... I don't think we should do this here. Can you poke Ionut about this one, it should be done on C++ side I guess or inside UnityRayTracingFetchVertexAttribute4

Reply:
I kinda of agree, but since the UnityRayTracingFetchVertexAttribute4(...) is generic, for all kinds of attributes, it makes sense to return 0 as a default value, rather than 1. What could be done maybe, is passing the result floatN as an attribute, and then return a boolean to indicate if the attribute was found or not?

Reply:
Ok there's a bunch of tests on the attribute data format already in UnityRayTracingFetchVertexAttribute4(...), so I guess we could add one on the attribute type, when returning the value if not found. I'll check with Ionut.

Reply:
Discussion ongoing with Ionut, in the meantime we could still merge this (and I added a FIXME comment) until we address this on the engine side.

Reply:
I am good with the fixme for now. And see what Ionut come with.